src/code/sourceLocationTracker.js

```diff
@@ -0,0 +1,55 @@
+'use strict'
+var EventManager = require('../lib/eventManager')
+var util = require('../helpers/global')
+var helper = require('../helpers/traceHelper')
+var SourceMappingDecoder = require('../util/sourceMappingDecoder')
+
+/**
+ * Process the source code location for the current executing bytecode
+ */
+function SourceLocationTracker (_codeManager) {
+  this.codeManager = _codeManager
+  util.extend(this, new EventManager())
```

Comment:
yes this is how it works in browser solidity but I did not updated the remix repository with the change we've done in the event manager. I will do that separately
We might need to find a way to not duplicate library code we use both in remix and browsersol.
we could put this in ethereumjs-util but not sure it makes sense cause the event manager is not ethereum related.